luigi/contrib/pyspark_runner.py

```diff
@@ -36,11 +36,66 @@
 import sys
 import os
 
+from luigi import configuration
+
 # this prevents the modules in the directory of this script from shadowing global packages
 sys.path.append(sys.path.pop(0))
 
 
-class PySparkRunner(object):
+class SparkEntryPoint(object):
+    def __init__(self, conf):
+        self.conf = conf
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
Aren't you overriding these anyway? Do you need this class? Can't you just do the `self.conf = conf` in both of the two classes below?

Reply:
surely I could, but I think that having `SparkEntryPoint` as an abstract class will add some clarity